app/code/community/Mastercard/Mpgs/Model/MpgsApi/Rest.php

```diff
@@ -321,6 +339,8 @@ public function authorizeFromSession(Mage_Sales_Model_Order $order)
         }
 
         $txnId = uniqid(sprintf('%s-', $orderId));
```

Comment:
Thanks @mdhilwan.
My comment here is meant to span across all these types of issues.

We had a talk internally and then also with @jk00000 if there is any point in making the change to uniqid.

For now - we conclude that it's really not necessary - since we do not expect a globally unique value from uniqid anyway. You can see that it's always used together with some other parameter.
It might even be a bit risky in some ways where uniqid value is re-used in some other request which has field with a low character count.

If you agree, then please mark the conversation resolved.

Reply:
@OnTapKristjan Orite as long as you and your team are aware and are agreeable 👍